test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateSaveResultTests.cs

```diff
@@ -0,0 +1,556 @@
+            }
+
+            MemoryStream stream = (MemoryStream)(odataRequestMessageWrapper.CachedRequestStream.Stream);
+            stream.Position = 0;
+
+            string payload = (new StreamReader(stream)).ReadToEnd();
+            payload.Should().Be("{\"@context\":\"http://www.odata.org/service.svc/$metadata#Cars/$delta\",\"value\":[{\"ID\":100,\"Owners@delta\":[{\"ID\":200,\"Name\":\"Edge\"}],\"Manufacturers@delta\":[{\"ID\":1001,\"Name\":\"Man-A\"}]}]}");
+        }
+
+        [Fact]
+        public void BulkUpdateGraphWithOneTopLevelElement_CreatedSuccessfully()
+        {
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+
+            this.context.AttachTo("Persons", person);
+            this.context.AttachTo("Cars", car1);
+            this.context.AttachTo("Cars", car2);
+
+            this.context.AddLink(person, "Cars", car1);
+            this.context.AddLink(person, "Cars", car2);
+
+            this.bulkUpdateGraph = this.GetBulkUpdateGraph(person);
+
+            Assert.True(this.bulkUpdateGraph != null);
+            Assert.Equal(2, this.bulkUpdateGraph.GetRelatedDescriptors(this.bulkUpdateGraph.TopLevelDescriptors.FirstOrDefault()).Count);
+        }
+
+        [Fact]
+        public void BulkUpdateGraphWithTwoTopLevelElement_CreatedSuccessfully()
+        {
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var person2 = new Person
+            {
+                ID = 200,
+                Name = "Edge"
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+
+            this.context.AttachTo("Persons", person);
+            this.context.AttachTo("Persons", person2);
+            this.context.AttachTo("Cars", car1);
+            this.context.AttachTo("Cars", car2);
+
+            this.context.AddLink(person, "Cars", car1);
+            this.context.AddLink(person, "Cars", car2);
+
+            this.bulkUpdateGraph = this.GetBulkUpdateGraph(person, person2);
+
+            Assert.True(this.bulkUpdateGraph != null);
+            Assert.Equal(2, this.bulkUpdateGraph.GetRelatedDescriptors(this.bulkUpdateGraph.TopLevelDescriptors.FirstOrDefault()).Count);
+        }
+
+        [Fact]
+        public void SerializeODataIdWithSomePropertiesToUpdate()
+        {
+            var car = new Car
+            {
+                ID = 100,
+            };
+
+            var persons = new List<Person>
+            {
+                new Person{
+                    ID = 200,
+                    Name = "Edge",
+                }
+            };
+
+            this.context.AttachTo("Persons", persons[0]);
+            DataServiceCollection<Person> personsCollection = new DataServiceCollection<Person>(this.context);

```

Comment:
Docs needed: We shouldn't forget to add that in `property tracking` we should use `DataServiceCollection`

Reply:
isn't this already documented?